File: packages/shared/src/types/registry.ts
export type DataQualityStatus =
  | "verified"
  | "partially_verified"
  | "needs_verification"
  | "degraded"
  | "unavailable";

export type SourceLevel = "federal" | "state" | "regional" | "county" | "municipal";

export type WaterManagementDistrict =
  | "NWFWMD"
  | "SRWMD"
  | "SJRWMD"
  | "SWFWMD"
  | "SFWMD";

export interface SourceRegistryEntry {
  id: string;
  name: string;
  agency: string;
  level: SourceLevel;
  category: string;
  access_method: string;
  base_url: string;
  docs_url: string;
  auth_required: boolean;
  license: string;
  refresh_cadence: string;
  data_quality_status: DataQualityStatus;
  verified_at: string | null;
  notes: string;
}

export interface CountyRegistryEntry {
  fips: string;
  name: string;
  state: "FL";
  region: string;
  water_management_districts: WaterManagementDistrict[];
  gis_open_data_url: string | null;
  procurement_url: string | null;
  permits_url: string | null;
  parcels_url: string | null;
  access_method: string | null;
  license: string | null;
  refresh_cadence: string;
  schema_notes: string | null;
  data_quality_status: DataQualityStatus;
  verified_at: string | null;
}

File: apps/api/app/schemas/registry.py
"""Registry passthrough schemas, mirroring packages/shared/src/types/registry.ts."""

from __future__ import annotations

from pydantic import BaseModel

DataQualityStatus = str
SourceLevel = str
WaterManagementDistrict = str


class SourceRegistryEntry(BaseModel):
    id: str
    name: str
    agency: str
    level: SourceLevel
    category: str
    access_method: str
    base_url: str
    docs_url: str
    auth_required: bool
    license: str
    refresh_cadence: str
    data_quality_status: DataQualityStatus
    verified_at: str | None
    notes: str


class CountyRegistryEntry(BaseModel):
    fips: str
    name: str
    state: str
    region: str
    water_management_districts: list[WaterManagementDistrict]
    gis_open_data_url: str | None
    procurement_url: str | None
    permits_url: str | None
    parcels_url: str | None
    access_method: str | None
    license: str | None
    refresh_cadence: str
    schema_notes: str | None
    data_quality_status: DataQualityStatus
    verified_at: str | None
